home_first_aid_kit/lib/home_first_aid_kit.ex

```diff
@@ -0,0 +1,45 @@
+
+defmodule HomeFirstAidKit do
+  @moduledoc """
+  Documentation for `HomeFirstAidKit`.
+  """
+
+  @doc """
+  Add medicament to First-Aid Kit with increase amount
+  """
+  def add_medicament(%Medicament{name: name, amount: amount}) do
+    storage = HomeFirstAidKitStorage.load()
+    existing = Map.get(storage, name, %Medicament{name: name, amount: 0})
+    new_medicament = Map.update(existing, :amount, 0, fn c_amount -> c_amount + amount end)
+    storage = Map.put(storage, name, new_medicament)
+    HomeFirstAidKitStorage.save(storage)
+    {:ok, new_medicament}
+  end
+
+  @doc """
+  Take medicament to First-Aid Kit with decrease amount
+  """
+  def use_medicament(%{name: name, amount: amount}) do
+    storage = HomeFirstAidKitStorage.load()
+    value = Map.get(storage, name)
+    case value do
+       nil -> {:empty, nil}
+       existing when is_integer(existing.amount) and existing.amount < amount ->
```

Comment:
```suggestion
       %{amount: ex_amnt} = existing when is_integer(ex_amnt) and ex_amnt < amount ->
```
Иначе тут можно словить исключение, если в `existing` мусор.